Directory.Packages.props

```diff
@@ -48,15 +46,18 @@
     <PackageVersion Include="OpenTracing" Version="[0.12.1,0.13)" />
 
     <!--
-        Typically, the latest stable version of System.Diagnostics.DiagnosticSource should be used here because:
-        1) Each major version bump will likely have some new OpenTelemetry capabilities (e.g. .NET 6 introduced Meter
-          API, .NET 7 added UpDownCounter, .NET 8 added Meter/Instrument level attributes support, .NET 9 added
-          Advice/Hint API, etc.).
-        2) Each minor version bump is normally security hotfixes or critical bug fixes.
-        3) The .NET runtime team provides extra backward compatibility guarantee to System.Diagnostics.DiagnosticSource
-          even during major version bumps, so compatibility is not a concern here.
+        For modern .NET runtimes, match the major version of packages to the major version of the runtime.
+        This is to ensure that the NuGet packages that users deploy match the support lifetimes of the .NET version they are using (i.e. LTS vs STS).
+        See https://github.com/open-telemetry/opentelemetry-dotnet/issues/5973 for more details.
     -->
-    <PackageVersion Include="System.Diagnostics.DiagnosticSource" Version="$(LatestRuntimeOutOfBandVer)" />
+    <PackageVersion Include="System.Diagnostics.DiagnosticSource" Version="$(RuntimePackageVersions)" />
+    <!--
+        However, for .NET 8 we need to use the .NET 9 package as the following APIs have been used to expose user-facing functionality:
+        - System.Diagnostics.Activity.ctor(string, string, IEnumerable<KeyValuePair<string, object>>)
+        - System.Diagnostics.Activity.AddException()
+        - System.Diagnostics.Activity.AddLink()
```

Comment:
Maybe this could be phrased to indicate that we need to (and can) use the latest version.

We, most likely, can use 10.0.x for all targets, like we can use 9.0.x.

For that matter, I'd use a property specific for this package's version.

Reply:
Depends on the resolution to this comment:

> Moving forward to preserve the intention of this PR we would need to accept that new functionality would be gated to upgrading to the matching version of .NET. For example, a new feature added in .NET 10 would require your application to net10.0 to use it.
>
> Alternatively we could take an approach where we only expose new APIs to older runtimes when the version is itself an LTS version.

The PR as-is is working on the assumption new things need newer TFMs/versions, so the MSBuild as-written is correct as it would only have an exception for `net8.0` and code would need to use `#if NET10_0_OR_GREATER`-like constructs as appropriate for future API additions.